### PR TITLE
fix: bug fixes user restructuring

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
fixing the chevron not getting toggled on outside click
showing label instead of value for org, merchant, profile switch dropdowns
fix profile id issue -  profiles not showing in dropdown
fix add new profile endpoint change
fix org dropdown width issue

image

## Motivation and Context
fix bugs

## How did you test it?
chevron is toggling as expected
label shown instead of id on dropdowns value

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [ ] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by PritishBudhiraja — APPROVED** [review]

File: src/entryPoints/HyperSwitchApp.res [hunk 1]
```diff
@@ -155,10 +155,12 @@ let make = () => {
...
                         headerActions={<div className="relative flex items-center gap-4 my-2 ">
                           <GlobalSearchBar />
-                          <SwitchMerchant
-                            userRole={userRole}
-                            isAddMerchantEnabled={userRole === "org_admin" ? true : false}
-                          />
+                          <RenderIf condition={!featureFlagDetails.userManagementRevamp}>
+                            <SwitchMerchant
+                              userRole={userRole}
+                              isAddMerchantEnabled={userRole === "org_admin" ? true : false}
```
Comment (PritishBudhiraja) [inline src/entryPoints/HyperSwitchApp.res:161]:
```suggestion
                              isAddMerchantEnabled={userRole === "org_admin"}
```

File: src/components/SelectBox.res [hunk 2]
```diff
@@ -1609,6 +1610,15 @@ module BaseDropdown = {
       setShowDropDown(_ => false)
       hasApplyButton ? newInputSelect.onChange(preservedAppliedOptions) : ()
     })
+
+    React.useEffect(() => {
+      let _ = switch toggleChevronState {
+      | Some(fn) => fn()
+      | None => ()
+      }
```
Comment (PritishBudhiraja) [inline src/components/SelectBox.res:1618]:
```suggestion
      switch toggleChevronState {
      | Some(fn) => fn()
      | None => ()
      }
```

File: src/screens/OMPSwitch/ProfileSwitch.res [hunk 3]
```diff
@@ -1,20 +1,17 @@
...
-    // let {profileId} = React.useContext(UserInfoProvider.defaultContext)
-
-    <div className="flex flex-col items-end gap-2" onClick={_ => setArrow(prev => !prev)}>
+  let make = (~currProfile, ~arrow) => {
+    <div className="flex flex-col items-end gap-2">
       <div
         className="flex items-center justify-end text-sm text-center text-black font-medium rounded hover:bg-opacity-80 bg-white w-fit">
         <div className="flex flex-col items-start px-2 py-2">
-          <p className="fs-10 text-nowrap"> {"current_profile_id"->React.string} </p>
+          <p className="fs-10 text-nowrap"> {{currProfile}->React.string} </p>
```
Comment (PritishBudhiraja) [inline src/screens/OMPSwitch/ProfileSwitch.res:8]:
```suggestion
          <p className="fs-10 text-nowrap"> {currProfile->React.string} </p>
```

File: src/screens/APIUtils/APIUtils.res [hunk 4]
```diff
@@ -453,13 +453,18 @@ let useGetURL = () => {
 
     /* BUSINESS PROFILE */
     | BUSINESS_PROFILE =>
-      switch id {
-      | Some(id) => `account/${merchantId}/business_profile/${id}`
-      | None =>
+      switch methodType {
+      | Get =>
         switch (userEntity, userManagementRevamp) {
         | (#Merchant, true) | (#Profile, true) => `account/${merchantId}/profile`
```
Comment (Riddhiagrawal001) [inline src/screens/APIUtils/APIUtils.res:459]: Add | (#Organization, true) fro checking

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**